eq-author-api/middleware/loadQuestionnaire.js

```diff
@@ -2,14 +2,22 @@ const { getQuestionnaire } = require("../utils/datastore");
 
 module.exports = async (req, res, next) => {
   const questionnaireId = req.header("questionnaireId");
-  if (questionnaireId) {
-    req.questionnaire = await getQuestionnaire(questionnaireId);
-    if (!req.questionnaire.metadata) {
-      req.questionnaire.metadata = [];
-    }
-    if (!req.questionnaire.sections) {
-      req.questionnaire.sections = [];
-    }
+  if (!questionnaireId) {
+    next();
+    return;
+  }
+  const questionnaire = await getQuestionnaire(questionnaireId);
+  req.questionnaire = questionnaire;
+  if (!questionnaire) {
+    next();
```

Comment:
Yeah, was hoping to push this down into resolvers as discussed. But I agree this needs tech debt discussion as it will have impact on other middleware and how we implement permissions etc.